src/main/java/duke/Command.java

```diff
@@ -0,0 +1,120 @@
+            }
+            return ui.list(list);
+        case "done":
+            int doneNum = sc.nextInt() - 1;
+            try {
+                tasks.get(doneNum).markAsDone();
+                return ui.done(tasks.get(doneNum));
+            } catch (IndexOutOfBoundsException e) {
+                return ui.showDoneError();
+            }
+        case "todo":
+            try {
+                String todoDescription = sc.nextLine().trim();
+                Task todo = new Todo(todoDescription);
+                if (todoDescription.isEmpty()) {
+                    throw new DukeException(ui.emptyDescriptionError());
+                }
+                tasks.add(todo);
+                return ui.todo(tasks, todo);
+            } catch (DukeException e) {
+                return e.getMessage();
+            }
+        case "delete":
+            int delNum = sc.nextInt() - 1;
+            try {
+                Task delete = tasks.get(delNum);
+                tasks.remove(delNum);
+                return ui.delete(tasks, delete);
+            } catch (IndexOutOfBoundsException e) {
+                return ui.showDeleteError();
+            }
+        case "deadline":
+            try {
+                String[] deadlineArr = sc.nextLine().split("/by");
+                if (deadlineArr[0].strip().isEmpty()) {
+                    throw new DukeException(ui.emptyDescriptionError());
+                }
+                LocalDate d1 = LocalDate.parse(deadlineArr[1].trim());
+                Task deadline = new Deadline(deadlineArr[0].trim(),
+                        d1.format(DateTimeFormatter.ofPattern("MMM dd YYYY")));
+                tasks.add(deadline);
+                return ui.deadline(tasks, deadline);
+            } catch (DateTimeParseException e) {
+                ui.showDeadlineError();
+                break;
+            } catch (DukeException e) {
+                return e.getMessage();
+            }
+        case "event":
+            try {
+                String[] eventArr = sc.nextLine().split("/at");
+                if (eventArr[0].strip().isEmpty()) {
+                    throw new DukeException(ui.emptyDescriptionError());
+                }
+                Task event = new Event(eventArr[0].trim(), eventArr[1].trim());
+                tasks.add(event);
+                return ui.event(tasks, event);
+            } catch (DukeException e) {
+                return e.getMessage();
+            }
+        case "find":
+            try {
+                String keyword = sc.nextLine().trim();
+                String findList = "\n";
+                if (keyword.strip().isEmpty()) {
+                    throw new DukeException(ui.emptyDescriptionError());
+                }
+                for (int i = 0; i < tasks.size(); i++) {
+                    if (tasks.get(i).getDescription().contains(keyword)) {
+                        findList += tasks.get(i) + "\n";
+                    }
+                }
```

Comment:
I feel that you can consider abstracting this inside the `TaskList` class itself